activejob/lib/active_job/logging.rb

```diff
@@ -87,12 +88,25 @@ def queue_name(event)
         def args_info(job)
           if job.arguments.any?
             ' with arguments: ' +
-              job.arguments.map { |arg| arg.try(:to_global_id).try(:to_s) || arg.inspect }.join(', ')
+              job.arguments.map { |arg| format(arg).inspect }.join(', ')
           else
             ''
           end
         end
 
+        def format(arg)
+          case arg
+          when Hash
+            arg.transform_values { |value| format(value) }
+          when Array
+            arg.map { |value| format(value) }
+          when GlobalID::Identification
+            begin arg.to_global_id rescue arg end
```

Comment:
`to_global_id` doesn't necessarily raise for an unpersisted model, just one that doesn't have an `id`. Also, the thing about `try` is that it just saves us from a `NoMethodError` when an object doesn't respond to a method, but not when the method itself raises. So this is what I wrote to prevent any errors and ensure that the arguments are always logged.